lib/index.js

```diff
@@ -528,29 +537,36 @@ class Generator extends EventEmitter {
       options
     );
 
-    try {
-      const Generator = require(modulePath); // eslint-disable-line import/no-dynamic-require
-      Generator.resolved = require.resolve(modulePath);
-      Generator.namespace = this.env.namespace(modulePath);
-      generator = this.env.instantiate(Generator, {
-        options,
-        arguments: options.arguments
-      });
-    } catch (err) {
-      if (err.code === 'MODULE_NOT_FOUND') {
-        generator = this.env.create(modulePath, {
-          options,
-          arguments: options.arguments
-        });
-      } else {
-        throw err;
+    if (typeof generator === 'string') {
+      try {
+        const Generator = require(generator); // eslint-disable-line import/no-dynamic-require
+        instantiatedGenerator = instantiate(Generator, generator);
+      } catch (err) {
+        if (err.code === 'MODULE_NOT_FOUND') {
+          instantiatedGenerator = this.env.create(generator, {
+            options,
+            arguments: options.arguments
+          });
+        } else {
+          throw err;
+        }
       }
+    } else {
+      assert(
+        generator.Generator,
+        'When passing an object to Generator#composeWith include the generator class to run in the Generator property'
```

Comment:
Let's add a code example to both error message here to make it super clear:

```js
When passing an object to Generator#composeWith include the generator class to run in the Generator property

this.composeWith({
    Generator: require('generator-x'),
    ...
})
```

(and let's color missing config line with `chalk.red()` to make it stand out)

Reply:
Changed the error, this is how it looks now on a console:
![image](https://user-images.githubusercontent.com/13201978/50363405-35369880-054a-11e9-9a6a-0336fd9ca897.png)